lib/iex/lib/iex/autocomplete.ex

```diff
@@ -336,8 +372,42 @@ defmodule IEx.Autocomplete do
     format_hint(name, hint)
   end
 
+  defp to_hint(%{kind: :map_key, name: name, value_is_map: true}, hint) when name == hint do
+    format_hint(name, hint) <> "."
+  end
+
+  defp to_hint(%{kind: :map_key, name: name}, hint) do
+    format_hint(name, hint)
+  end
+
   defp format_hint(name, hint) do
     hint_size = byte_size(hint)
     :binary.part(name, hint_size, byte_size(name) - hint_size)
   end
+
+  defp get_evaluator do
+    case IEx.Server.local do
+      nil -> nil
+      pid ->
+        {:dictionary, dictionary} = Process.info(pid, :dictionary)
+        dictionary[:evaluator]
+    end
+  end
+
+  defp get_variable_value(ast_node, evaluator) do
+    binding = IEx.Evaluator.get_binding(evaluator)
+
+    {_, value} = Macro.postwalk(ast_node, {:binding, binding}, fn
```

Comment:
Hrm, that's too generate for what you are trying to build in the sense that I don't know if it will actually work or fail in other scenarios. My suggestion would be to convert `ast_node` to a list of variable accesses. And then use Enum.reduce. In pseudo code:

    {var, fields} = extract_var_and_fields(ast_node)
    value = IEx.Evaluator.binding(evaluator, var)
    Enum.reduce fields, {:ok, value}, fn
      field, {:ok, acc} when is_map(acc) -> Map.fetch(acc, field)
      _field, _acc -> :error
    end)